### DISCUSSION TITLE
How to train the special element with different atoms and structrure with deepmd-kit?

### DISCUSSION THREAD

**Discussion by zcb-code** [deepmodeling/deepmd-kit#486] [Q&A] [answered]
I am training an element PD, I have many structures with different atom numbers. I have many OUTCARs.  Is it possible to use these OUTCARS(only with one step self-consistent computing), to train a initial *pd.file ? After I run the script:

from dpdata import LabeledSystem,MultiSystems
from glob import glob
"""
process multi systems
"""
fs=glob('./*/OUTCAR')  # remeber to change here !!!
ms=MultiSystems()
for f in fs:
    try:
        ls=LabeledSystem(f)
    except:
        print(f)
    if len(ls)>0:
        ms.append(ls)

ms.to_deepmd_raw('deepmd')
ms.to_deepmd_npy('deepmd')
__________________________________________________________________________________________________________________________________
$ cd deepmd;tree, I got:
.
|-- Li10
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
|-- Li4
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
|-- Li5
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
|-- Li6
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
|-- Li7
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
|-- Li8
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
|-- Li9
|   |-- box.raw
|   |-- coord.raw
|   |-- energy.raw
|   |-- force.raw
|   |-- set.000
|   |   |-- box.npy
|   |   |-- coord.npy
|   |   |-- energy.npy
|   |   |-- force.npy
|   |   `-- virial.npy
|   |-- type.raw
|   |-- type_map.raw
|   `-- virial.raw
`-- out

__________________________________________________________________________________________________________________________________

My second question is that, if possible, how do I modify my Li.json file before train?
my Li.json :
{
    "_comment": " model parameters",
    "model": {
        "type_map":     ["Li"],
        "descriptor" :{
            "type":             "se_a",
            "sel":              [],
            "rcut_smth":        0.50,
            "rcut":             6.00,
            "neuron":           [25, 50, 100],
            "resnet_dt":        false,
            "axis_neuron":      16,
            "seed":             1,
            "_comment":         " that's all"
        },
        "fitting_net" : {
            "n_neuron":         [240, 240, 240],
            "resnet_dt":        true,
            "seed":             1,
            "_comment":         " that's all"
        },
        "_comment":     " that's all"
    },

    "learning_rate" :{
        "type":         "exp",
        "start_lr":     0.001,
        "decay_steps":  500,
        "decay_rate":   0.95,
        "_comment":     "that's all"
    },

    "loss" :{
        "start_pref_e": 0.02,
        "limit_pref_e": 1,
        "start_pref_f": 1000,
        "limit_pref_f": 1,
        "start_pref_v": 0,
        "limit_pref_v": 0,
        "_comment":     " that's all"
    },

    "_comment": " traing controls",
    "training" : {
        "systems":      ["./"],     
        "set_prefix":   "set",
        "stop_batch":   500000,
        "batch_size":   1,

        "seed":         1,

        "_comment": " display and restart",
        "_comment": " frequencies counted in batch",
        "disp_file":    "lcurve.out",
        "disp_freq":    100,
        "numb_test":    10,
        "save_freq":    1000,
        "save_ckpt":    "model.ckpt",
        "load_ckpt":    "model.ckpt",
        "disp_training":true,
        "time_training":true,
        "profiling":    false,
        "profiling_file":"timeline.json",
        "_comment":     "that's all"
    },

    "_comment":         "that's all"

__________________________________________________________________________________________________________________________________
how to modify the     "systems":      ["./"],      part? 

Thanks.

**Comment by amcadmus** [comment]
RTD https://deepmd.readthedocs.io/en/master/train-input.html

**Reply by zcb-code** [reply]
Thanks for your recommend.

**Comment by njzjz** [comment, marked as answer]
> how to modify the "systems": ["./"], part?

Something like `["data/Li1", "data/Li2", "data/Li3"]`. I suggest to write a script to generate it.

**Reply by zcb-code** [reply]
Thanks for your kind replay. I have another two question, how many  sets are needed in files of Li2? And the set number in Li2,Li3 needs to equal or not?

**Reply by njzjz** [reply]
It depends on you. You may see https://doi.org/10.6084/m9.figshare.12973055 for an example.

**Reply by zcb-code** [reply]
Thanks for your useful reply.

**Comment by njzjz** [comment]
@zcb-code Note: if you are using either v1.3.0, v1.3.1, v1.3.2, or v1.3.3, the key `model/fitting_net/n_neuron` will not work due to a bug. Use `model/fitting_net/neuron` instead.
